otelcol/collector.go

```diff
@@ -226,6 +231,16 @@ func (col *Collector) DryRun(ctx context.Context) error {
 // Run starts the collector according to the given configuration, and waits for it to complete.
 // Consecutive calls to Run are not allowed, Run shouldn't be called once a collector is shut down.
 func (col *Collector) Run(ctx context.Context) error {
+	col.runWg.Add(1)
+	defer col.runWg.Done()
```

Comment:
I think you create a deadlock here https://github.com/open-telemetry/opentelemetry-collector/pull/8811/files#diff-3c058aa5dde6d9b2b413b65dce28fe97e9cf859a0f03629639843aac6bc26e2aR286

Reply:
havent run into a deadlock but i will for sure verify

Reply:
yeah i think that's right. will need to figure something else

Reply:
still looking at it at first i thought `Shutdown` is called within `Run` but this is not the case so this deadlock is not there.
i'm drawing some flow charts and it should be fine from what i'm seeing. @bogdandrutu  can you elaborate on where you think the deadlock is?

changes are only to exported funcs which are not calling each other
- when Runs are running Shutdown waits for them
- when they are not waitgroup does not block
- and each run should finish as `.shutdown` is triggered from Shutdown which escapes run loop and decrement waitgroup counter.